pkg/web/router.go

```diff
@@ -138,16 +143,20 @@ func NewRouter(config *RouterConfig, wpool pool.WorkerPoolInterface) http.Handle
 	}
 
 	// register endpoints as last step
-	statusAPI.RegisterEndpoints(router)
-	metricsAPI.RegisterEndpoints(router)
-
-	identitiesAPI.RegisterEndpoints(router)
-	clientsAPI.RegisterEndpoints(router)
-	idpAPI.RegisterEndpoints(router)
-	schemasAPI.RegisterEndpoints(router)
-	rulesAPI.RegisterEndpoints(router)
-	rolesAPI.RegisterEndpoints(router)
-	groupsAPI.RegisterEndpoints(router)
+	statusAPI.RegisterEndpoints(apiRouter)
+	metricsAPI.RegisterEndpoints(apiRouter)
+
+	identitiesAPI.RegisterEndpoints(apiRouter)
+	clientsAPI.RegisterEndpoints(apiRouter)
+	idpAPI.RegisterEndpoints(apiRouter)
+	schemasAPI.RegisterEndpoints(apiRouter)
+	rulesAPI.RegisterEndpoints(apiRouter)
+	rolesAPI.RegisterEndpoints(apiRouter)
+	groupsAPI.RegisterEndpoints(apiRouter)
+
+	uiAPI.RegisterEndpoints(router)
+
+	router.Mount("/api", apiRouter)
```

Comment:
this works and was the part I overlooked in my first comment 

albeit being ok and just right it triggers quite a change in the rest of the code which I would probably spare

on a side note, we would also need to have a catch all to point to the UI handler and we could use https://pkg.go.dev/github.com/go-chi/chi/v5#Mux.NotFound
on top of still having the UI served from `/ui`